crates/sui-types/src/zk_login_authenticator.rs

```diff
@@ -89,11 +88,32 @@ impl Hash for ZkLoginAuthenticator {
 }
 
 impl AuthenticatorTrait for ZkLoginAuthenticator {
-    fn verify_user_authenticator_epoch(&self, epoch: EpochId) -> SuiResult {
-        // Verify the max epoch in aux inputs is <= the current epoch of authority.
+    fn verify_user_authenticator_epoch(
+        &self,
+        epoch: EpochId,
+        upper_bound_max_epoch: Option<EpochId>,
+    ) -> SuiResult {
+        // the checks here ensure that `current_epoch + upper_bound_max_epoch >= self.max_epoch >= current_epoch`.
+        // 1. if the config for upper bound is set, ensure that the max epoch in signature is not larger than epoch + upper_bound.
+        if let Some(upper_bound) = upper_bound_max_epoch {
+            if self.get_max_epoch() > epoch + upper_bound {
+                return Err(SuiError::InvalidSignature {
+                    error: format!(
+                        "ZKLogin max epoch too large {}, current epoch {}",
```

Comment:
The error message should include `upper_bound` either directly, or indirectly, so that the user can figure out what the maximum allowable value is

Reply:
good point, revised